app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java

```diff
@@ -210,31 +227,22 @@ protected void onPostCreate(Bundle savedInstanceState) {
     protected void onResume() {
         super.onResume();
 
-        MessagePubSub.NewMessagesSubscriber subscriber = new MessagePubSub.NewMessagesSubscriber() {
-            @Override
-            public void onNewMessage(final Message msg) {
-                if (msg.getSenderId().equals(NetworkUtil.getMac())) {
-                    //Don't notify about incoming messages from self
-                    return;
-                }
-                MainActivity.this.runOnUiThread(new Runnable() {
-                    @Override
-                    public void run() {
-                        ShowMessageDialogFragment.showNewMessages(getFragmentManager(), msg);
-                    }
-                });
-            }
-        };
-        MessagePubSub.getInstance().subscribe(subscriber);
+        // Register to receive messages.
+        // We are registering an observer
+        LocalBroadcastManager.getInstance(this).registerReceiver(
```

Comment:
can also be a static method at MessageBroadcastReceiver:
register(MessageBroadcastReceiver mbr)